src/dash-playlist-loader.js

```diff
@@ -336,35 +387,42 @@ export default class DashPlaylistLoader extends EventTarget {
     });
   }
 
+  haveMaster_() {
+    this.state = 'HAVE_MASTER';
+    // clear media request
+    this.mediaRequest_ = null;
+
+    if (!this.masterPlaylistLoader_) {
+      this.master = this.parseMasterXml();
+      // We have the master playlist at this point, so
+      // trigger this to allow MasterPlaylistController
+      // to make an initial playlist selection
+      this.trigger('loadedplaylist');
+    } else if (!this.media_) {
+      // no media playlist was specifically selected so select
+      // the one the child playlist loader was created with
+      this.media(this.childPlaylist_);
+    }
+  }
+
   /**
    * Handler for after client/server clock synchronization has happened. Sets up
    * xml refresh timer if specificed by the manifest.
    */
   onClientServerClockSync_() {
-    this.master = this.parseMasterXml();
+    this.haveMaster_();
 
-    this.state = 'HAVE_MASTER';
-
-    this.trigger('loadedplaylist');
-
-    if (!this.media_) {
-      // no media playlist was specifically selected so start
-      // from the first listed one
+    if (!this.hasPendingRequest() && !this.media_) {
       this.media(this.master.playlists[0]);
     }
-    // trigger loadedmetadata to resolve setup of media groups
-    // trigger async to mimic behavior of HLS, where it must request a playlist
-    window.setTimeout(() => {
-      this.trigger('loadedmetadata');
-    }, 0);
 
     // TODO: minimumUpdatePeriod can have a value of 0. Currently the manifest will not
     // be refreshed when this is the case. The inter-op guide says that when the
     // minimumUpdatePeriod is 0, the manifest should outline all currently available
     // segments, but future segments may require an update. I think a good solution
     // would be to update the manifest at the same rate that the media playlists
     // are "refreshed", i.e. every targetDuration.
-    if (this.master.minimumUpdatePeriod) {
+    if (this.master && this.master.minimumUpdatePeriod) {
```

Comment:
Mostly just curious was there a situation where `this.master` was `undefined` and you needed to add this check? or just adding it in the name of good practices? (which I don't disagree with)

Reply:
At this point, it's just good practices. I think it was actually necessarily before I stopped childLoaders from calling `onClientServerClockSync_`